BUG: Failure to build on windows: `'ssize_t': undeclared identifier`

Ever since the merging of https://github.com/numpy/numpy/pull/21025 I've been unable to locally build from the `main` branch when using windows.
The full build log is attached below, but it seems `ssize_t` is currently not defined in tokenize.cpp:

``` 
numpy\core\src\multiarray\textreading\tokenize.cpp(392): error C2065: 'ssize_t': undeclared identifier
numpy\core\src\multiarray\textreading\tokenize.cpp(392): error C2146: syntax error: missing ';' before identifier 'offset_last'
numpy\core\src\multiarray\textreading\tokenize.cpp(392): error C2065: 'offset_last': undeclared identifier
numpy\core\src\multiarray\textreading\tokenize.cpp(393): error C2065: 'ssize_t': undeclared identifier
numpy\core\src\multiarray\textreading\tokenize.cpp(393): error C2146: syntax error: missing ';' before identifier 'end_last'
numpy\core\src\multiarray\textreading\tokenize.cpp(393): error C2065: 'end_last': undeclared identifier
numpy\core\src\multiarray\textreading\tokenize.cpp(394): error C2065: 'end_last': undeclared identifier
numpy\core\src\multiarray\textreading\tokenize.cpp(394): error C2065: 'offset_last': undeclared identifier
```

Reproduce the code example:
------------------------------
``` shell
git clean -xfd
python setup.py build
```

Error message:
---------------
* Full build log: [build_log.txt](https://github.com/numpy/numpy/files/8081332/build_log.txt)

NumPy/Python version information:
-------------------------------------
* Numpy version: `main` branch (post-https://github.com/numpy/numpy/commit/67bb936e8d8bc29c1dc4e54449a173280d43b9dd).
* Python version: `3.10.2 | packaged by conda-forge | (main, Feb  1 2022, 19:27:13) [MSC v.1929 64 bit (AMD64)]`

--- a/numpy/core/src/multiarray/textreading/tokenize.cpp
+++ b/numpy/core/src/multiarray/textreading/tokenize.cpp
@@ -389,8 +389,8 @@ tokenize(stream *s, tokenizer_state *ts, parser_config *const config)
      */
     if (ts->num_fields == 1
             || ts->unquoted_state == TOKENIZE_UNQUOTED_WHITESPACE) {
-        ssize_t offset_last = ts->fields[ts->num_fields-1].offset;
-        ssize_t end_last = ts->fields[ts->num_fields].offset;
+        std::ptrdiff_t offset_last = ts->fields[ts->num_fields-1].offset;
+        std::ptrdiff_t end_last = ts->fields[ts->num_fields].offset;
         if (!ts->fields->quoted && end_last - offset_last == 1) {
             ts->num_fields--;
         }

--- a/numpy/core/src/multiarray/textreading/tokenize.cpp
+++ b/numpy/core/src/multiarray/textreading/tokenize.cpp
@@ -389,8 +389,8 @@ tokenize(stream *s, tokenizer_state *ts, parser_config *const config)
      */
     if (ts->num_fields == 1
             || ts->unquoted_state == TOKENIZE_UNQUOTED_WHITESPACE) {
-        std::ptrdiff_t offset_last = ts->fields[ts->num_fields-1].offset;
-        std::ptrdiff_t end_last = ts->fields[ts->num_fields].offset;
+        size_t offset_last = ts->fields[ts->num_fields-1].offset;
+        size_t end_last = ts->fields[ts->num_fields].offset;
         if (!ts->fields->quoted && end_last - offset_last == 1) {
             ts->num_fields--;
         }

